crates/webidl/src/util.rs

```diff
@@ -707,3 +712,49 @@ pub fn public() -> syn::Visibility {
         pub_token: Default::default(),
     })
 }
+
+/// When generating our web_sys APIs we default to setting slice references that
+/// get passed to JS as mutable in case they get mutated in JS.
+///
+/// In certain cases we know for sure that the slice will not get mutated - for
+/// example when working with the WebGlRenderingContext APIs.
+///
+/// Here we implement a whitelist for those cases. This whitelist is currently
+/// maintained by hand.
+fn maybe_adjust<'a>(mut idl_type: IdlType<'a>, id: &'a OperationId) -> IdlType<'a> {
+    let immutable_f32_fns: Vec<&'static str> = vec![
+        // WebGlRenderingContext
+        "uniform1fv",
+        "uniform2fv",
+        "uniform3fv",
+        "uniform4fv",
+        "uniformMatrix2fv",
+        "uniformMatrix3fv",
+        "uniformMatrix4fv",
+        "vertexAttrib1fv",
+        "vertexAttrib2fv",
+        "vertexAttrib3fv",
+        "vertexAttrib4fv",
+        // TODO: Add another type's functions here. Leave a comment header with the type name
+    ];
+
+    let mut immutable_f32_slice_whitelist: HashSet<&'static str> = HashSet::new();
+    for function_name in immutable_f32_fns.into_iter() {
+        immutable_f32_slice_whitelist.insert(function_name);
+    }
+
+        // example `op`... -> "vertexAttrib1fv"
+    if let OperationId::Operation(Some(op)) = id {
```

Comment:
This can reduce some indentation as:

```rust
let id = match id {
    OperationId::Operation(Some(op)) => op,
    _ => return idl_type,
};

if !immutable_whitelist.contains(id) {
    return idl_type;
}

// ...
```